src/tests/p11test/Makefile.am

```diff
@@ -23,7 +23,7 @@ p11test_SOURCES = p11test.c p11test_loader.c \
 	p11test_case_pss_oaep.c \
 	p11test_helpers.c
 p11test_CFLAGS = -DNDEBUG $(CMOCKA_CFLAGS)
-p11test_LDADD = -lssl -lcrypto $(CMOCKA_LDLAGS)
+p11test_LDADD = -lssl -lcrypto $(CMOCKA_LIBS)
```

Comment:
Would it make sense to use `$(OPTIONAL_OPENSSL_LIBS)` and/or to additionally check for libssl in `configure.ac`?

Reply:
Once we are in rewriting that, yes, we should use `OPTIONAL_OPENSSL_LIBS`. Looking thrugh it, libssl itself is not needed at all if I am right. Adjusting configure to build tests also only if we have openssl is also good idea. Will commit changes soon.